src/xdocs/config_imports.xml

```diff
@@ -1691,24 +1742,52 @@ import java.util.stream.IntStream;
   &lt;property name=&quot;sortStaticImportsAlphabetically&quot; value=&quot;true&quot;/&gt;
 &lt;/module&gt;
         </source>
+        <p>
+          Example:
+        </p>
+        <source>
+import static java.awt.Button.A; // Violation; Should follow 'sortStaticImportsAlphabetically'
+import static java.lang.Math.PI;
+import static java.lang.Math.abs; // Ok, alphabetical case sensitive ASCII order, 'P' &lt; 'a'
+import java.lang.Math.sqrt; // Ok, Follow property 'Option' value 'above'
+
+import static java.io.File.createTempFile;
+import java.io.File;
+
+import java.io.IOException; // Violation; Should follow 'Group' property
 
-        <p>To configure the check so that it matches default IntelliJ IDEA formatter
-           configuration (tested on v2018.2):</p>
+import org.albedo.*; // Violation; Should follow 'ordered' property *{comes after javax imports}
+
+import static javax.swing.WindowConstants.*;
+import javax.swing.JComponent;
+import org.apache.http.ClientConnectionManager; // Violation; Should follow 'separated' property
+
+import org.linux.apache.server.SoapServer; // Ok
```

Comment:
Using the above config with latest jar from https://checkstyle.sourceforge.io/:

```
➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="ImportOrder">
            <property name="groups" value="/^java\./,javax,org,com"/>
            <property name="ordered" value="true"/>
            <property name="separated" value="true"/>
            <property name="option" value="above"/>
            <property name="sortStaticImportsAlphabetically" value="true"/>
        </module>
    </module>
</module>
➜  src cat SomeClass.java 
import static java.awt.Button.A; // Violation; Should follow 'sortStaticImportsAlphabetically'
import static java.lang.Math.PI;
import static java.lang.Math.abs; // Ok, alphabetical case sensitive ASCII order, 'P' &lt; 'a'
import java.lang.Math.sqrt; // Ok, Follow property 'Option' value 'above'

import static java.io.File.createTempFile;
import java.io.File;

import java.io.IOException; // Violation; Should follow 'Group' property

import org.albedo.*; // Violation; Should follow 'ordered' property {comes after javax imports}

import static javax.swing.WindowConstants.*;
import javax.swing.JComponent;
import org.apache.http.ClientConnectionManager; // Violation; Should follow 'separated' property

import org.linux.apache.server.SoapServer; // Ok

import com.neurologic.http.HttpClient; // Ok
import com.neurologic.http.impl.ApacheHttpClient; // Ok

public class SomeClass { }
➜  src java -jar checkstyle-8.39-all.jar 
Missing required parameter: '<files>'
Usage: checkstyle [OPTIONS]... FILES...
Try 'checkstyle --help' for more information.
➜  src java -jar checkstyle-8.39-all.jar -c config.xml SomeClass.java
Starting audit...
[ERROR] /home/nick/Desktop/Tester/src/SomeClass.java:6:1: Extra separation in import group before 'java.io.File.createTempFile' [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/SomeClass.java:6:1: Wrong order for 'java.io.File.createTempFile' import. [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/SomeClass.java:9:1: Extra separation in import group before 'java.io.IOException' [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/SomeClass.java:13:1: Wrong order for 'javax.swing.WindowConstants.*' import. [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/SomeClass.java:15:1: 'org.apache.http.ClientConnectionManager' should be separated from previous imports. [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/SomeClass.java:17:1: Extra separation in import group before 'org.linux.apache.server.SoapServer' [ImportOrder]
Audit done.
Checkstyle ends with 6 errors.
```
Why do Checkstyle violations differ from your comments?

Reply:
in 15:1 Violation comments meaning is same and for 17:1: i wrote it considering 13,14,15 one group so next org group should be separated  that's why i leave a line (16)
for 13:1 i thought error here should be on 11:1 but i was wrong
No explanation for 6 and 1sorry  i will change them
i guess i have to test with config like this for code example.
Do i have to change the comments and write what we get in CheckStyle violation?

Reply:
`src` is simply the directory that I am in; please ignore that.  


> i guess i have to test with config like this for code example.

Yes, this is also good practice. [Please see my comment to you at](https://github.com/checkstyle/checkstyle/pull/9144#discussion_r553747380) #9144 .

Do not mark items as resolved; that is for the reviewer to do. Please "un"resolve the previous items.

Go though all examples and configs and make sure that all are consistent with Checkstyle behavior.

Reply:
okay

Reply:
done 
what's with  Wercker pipeline failed ?

Reply:
> what's with Wercker pipeline failed ?

https://github.com/checkstyle/checkstyle/issues/9176

It is unrelated to this PR.